src/MultipleValueAutocomplete/useMultipleValueAutocomplete.ts

```diff
@@ -0,0 +1,172 @@
+import {
+  useCombobox,
+  UseComboboxGetItemPropsOptions,
+  useMultipleSelection,
+} from "downshift";
+import { useCallback, useEffect, useMemo, useRef } from "react";
+
+import { SyntheticChangeEvent } from "../../types/utils";
+import { Choice } from "../Filter";
+import { useTextWidth } from "../tools/useTextWidth";
+import { mergeRefs } from "../utils/mergeRefs";
+
+export interface UseMultipleValueAutocomplete {
+  choices: Choice[];
+  enableReinitialize?: boolean;
+  name?: string;
+  initialValue: Choice[];
+  onChange?: (event: SyntheticChangeEvent<string[]>) => void;
+  onInputChange?: (value: string) => void;
+}
+
+function useMultipleValueAutocomplete({
+  choices,
+  enableReinitialize,
+  initialValue,
+  name,
+  onChange,
+  onInputChange,
+}: UseMultipleValueAutocomplete) {
+  const anchor = useRef<HTMLDivElement>();
+  const input = useRef<HTMLInputElement>();
+
+  const [inputWidth, setInputWidth] = useTextWidth(
```

Comment:
```suggestion
  const [inputWidth, setInputText] = useTextWidth(
```

`setInputWidth` is a bit confusing, since we're setting text content not width - I think `setInputText` would be better name